### DISCUSSION TITLE
API/Self  is taking too long time

### DISCUSSION THREAD

**Discussion by cherrycharan** [Budibase/budibase#7861] [Help]
The page loading time is very long , when the application makes a api call to API/Self from budibase-client.js 
attaching the screenshot 
<img width="427" alt="image" src="https://user-images.githubusercontent.com/16544980/191327557-a084fef7-4a28-4c88-8388-b00a41a1835a.png">

**Comment by melohagan** [comment]
Hey @cherrycharan 

Are you self-hosted?

**Reply by cherrycharan** [reply]
Yes , we are using self hosted one

**Reply by yeoldegrove** [reply]
I have the same issue...
e.g. there are 2 searches which each take about 2s and before that the call to `/api/self` which takes 4s. Sums up to 8s of page loading time. The 4s for the searches are explainable but what exactly happens at `/api/self` ?

![image](https://user-images.githubusercontent.com/8668070/199186331-eb30ea1d-06e7-4fe7-944a-750ba168a6ae.png)

**Reply by melohagan** [reply]
Hey @yeoldegrove Would you be able to provide your app export to support@budibase.com ?

I'm just wondering if there could be anything unusual about your app. Is it very large?

**Reply by yeoldegrove** [reply]
@melohagan Just tried to export it with "exclude rows". It is a 197 MB export but it still has too much sensible data in it to upload it anywhere. Without the "exclude rows" it is 207 MB.

Any other hints what to optimize or look for?

**Reply by melohagan** [reply]
Wow that's gigantic. How many screens and components are in your app?

**Reply by yeoldegrove** [reply]
There are about ~30 screens but each with a very limited set of components. Mostly displaying a table or changing a row.

Just updated to 2.0.40 (was on 2.0.21 before) and retried the export. This time I get a 40 MB `tar.gz` but in there is still the 207 MB `db.txt` including all our sensible data. I guess the "export without rows" still does not work :(
Or...If I take a deeper look inside the export...this actually might not be the rows but my automation logs that are this large???

```
      {
        "_id": "log_au_2022-10-31T09:30:09.087Z_au_4c8c9d42be574dfaa9e979eaec97354d_success",
        "_rev": "1-ea50f476ace542603843826ecb8d510f",
        "trigger": {
          "id": "D4oWCVDJx",
          "stepId": "CRON",
          "inputs": null,
          "outputs": {
            "timestamp": 1667208600043
          }
        },
        "steps": [
          {
            "id": "D4oWCVDJx",
            "stepId": "CRON",
            "inputs": null,
            "outputs": {
              "timestamp": 1667208600043
            }
          },
          {
            "id": "k0_DWepZq",
            "stepId": "QUERY_ROWS",
            "inputs": {
              "tableId": "ta_8e74217a8ba94817b640dfa4c0530e4c",
              "filters": {
                "string": {},
                "fuzzy": {},
                "range": {},
                "equal": {},
                "notEqual": {},
                "empty": {},
                "notEmpty": {},
                "contains": {},
                "notContains": {}
              },
              "filters-def": [],
              "sortColumn": "season",
              "sortOrder": "ascending",
              "limit": "10000000"
            },
            "outputs": {
              "rows": [
                {
                  "_id": "ro_ta_8e74217a8ba94817b640dfa4c0530e4c_01367aec9a5f4c4eae435f2e4a0dcf89",
                  "_rev": "35-bf4ff022f14e83cb5aa3ab4a151c8c2b",
```

Could we implement a switch to exclude these too?

**Reply by melohagan** [reply]
Ah the automation logs from CRON. How frequently does your cron run and how long has it been running?

The exclude rows does seem to be working okay. 207-197 = 10MB which is a lot of text and many rows!

**Reply by yeoldegrove** [reply]
I have a cron job exporting a table every 5 minutes which has been running for weeks.
The export is too large to analyze this quick...but for now it looks to me that at least a large portion is cron logs.
